app/Http/Controllers/SrdController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Srd\SrdRoute;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+
+class SrdController
+{
+    public function searchRoutes(Request $request): JsonResponse
+    {
+        // Do some validation
+        $rules = [
+            'origin' => 'required|alpha',
+            'destination' => 'required|alpha',
+            'requestedLevel' => 'integer',
+        ];
+
+        $validator = Validator::make($request->query(), $rules);
+        if ($validator->fails()) {
+            return response()->json(['errors' => $validator->errors()->all()], 400);
+        }
+
+        // Build the query
+        $query = SrdRoute::where('origin', $request->query('origin'))
```

Comment:
I think you can reference the fields directly from the validator so we are absolutely sure it is a sane request value, and that it exists.

Reply:
Done!